lib/plugins.ts

```diff
@@ -23,29 +23,35 @@ const migrateState = <S extends IState>(oldState: IState, newState: S): S => {
 export const LocalStoragePlugin = <S extends IState>(
   module: IGenericModule<S>,
 ): IPlugin<S> => {
+  const onDataChange = (value: S): S => {
+    const payload: ILocalStoragePluginState<S> = {
+      state: value,
+      __version: module.options.version,
+    }
+    localStorage.setItem(module.options.name, JSON.stringify(payload))
+    return value
+  }
   return {
     onStateInit: (state: S) => {
       const { name, version } = module.options
       const stateString = localStorage.getItem(name)
+      let initializedState: S
       if (!stateString) {
-        return state
-      }
-      const payload = JSON.parse(stateString) as ILocalStoragePluginState<S>
-      if (payload.__version !== version) {
-        console.warn(
-          `Upgrade module '${name}' from ${payload.__version} to ${version}.\n` +
-            'States has been migrated',
-        )
-        return migrateState(payload.state, state)
-      }
-      return payload.state
-    },
-    onDataChange: (value: S) => {
-      const payload: ILocalStoragePluginState<S> = {
-        state: value,
-        __version: module.options.version,
+        initializedState = state
+      } else {
+        const payload = (JSON.parse(stateString) || {}) as ILocalStoragePluginState<S>
+        if (payload.__version !== version) {
+          console.warn(
+            `Upgrade module '${name}' from ${payload.__version ?? 0} to ${version}.\n` +
+              'States has been migrated',
+          )
+          initializedState = migrateState(payload.state ?? {}, state)
```

Comment:
I noticed that if the app storage gets corrupted somehow, this can lead to a runtime error. It's probably really rare in practice, but I noticed it when I ran the example locally because I still had very old `example` app state in my browser.